site/nav.yml

```diff
@@ -21,6 +21,7 @@ nav:
     - Spark: spark-quickstart.md
     - Hive: hive-quickstart.md
   - Docs: 
+    - nightly: '!include docs/docs/nightly/mkdocs.yml'
```

Comment:
`nightly` and `latest` look similar and a bit confusing. I think we can merge `latest` and `1.5.0` to `1.5.0 (latest)` or `1.5.0 (stable)`.

Reply:
I'm open to it, I think this was what we went with in the last discussion, but happy to bring this up. See here: https://github.com/apache/iceberg/pull/9943#discussion_r1532828646

CC: @danielcweeks @RussellSpitzer any thoughts here?

Reply:
The label isn't what matters the most. What we care about are docs under `docs/latest/...` URLs. We can't really combine those with, for example, the 1.5.0 docs because we want to have the ability to permalink to a specific version if that's what you're viewing.